File: python-2/keyword_search.py
#!python2
from OnetWebService import OnetWebService
import sys

def get_user_input(prompt):
    result = ''
    while (len(result) == 0):
        result = raw_input(prompt + ': ').strip()
    return result

def check_for_error(service_result):
    if 'error' in service_result:
        sys.exit(service_result['error'])

username = get_user_input('O*NET Web Services username')
password = get_user_input('O*NET Web Services password')
onet_ws = OnetWebService(username, password)

vinfo = onet_ws.call('about')
check_for_error(vinfo)
print "Connected to O*NET Web Services version " + str(vinfo['api_version'])
print ""

kwquery = get_user_input('Keyword search query')
kwresults = onet_ws.call('online/search',
                         ('keyword', kwquery),
                         ('end', 5))
check_for_error(kwresults)
if (not 'occupation' in kwresults) or (0 == len(kwresults['occupation'])):
    print "No relevant occupations were found."
    print ""
else:
    print "Most relevant occupations for \"" + kwquery + "\":"
    for occ in kwresults['occupation']:
        print "  " + occ['code'] + " - " + occ['title']
    print ""

File: perl-5/keyword_search.pl
#!perl
use strict;
use warnings;

use Local::OnetWebService ();

sub get_user_input {
  my ($prompt) = @_;
  my $result = '';
  while (!length($result)) {
    print $prompt . ': ';
    $result = <STDIN>;
    chomp $result;
    $result =~ s/^\s+//;
    $result =~ s/\s+$//;
  }
  return $result;
}

sub check_for_error {
  my ($service_result) = @_;
  if (defined $service_result->{'error'}) {
    die $service_result->{'error'} . "\n";
  }
}

my $username = get_user_input('O*NET Web Services username');
my $password = get_user_input('O*NET Web Services password');
my $onet_ws = Local::OnetWebService->new($username, $password);

my $vinfo = $onet_ws->call('about');
check_for_error($vinfo);
print "Connected to O*NET Web Services version " . $vinfo->{'api_version'} . "\n\n";

my $kwquery = get_user_input('Keyword search query');
my $kwresults = $onet_ws->call('online/search',
                               'keyword' => $kwquery,
                               'end' => 5);
check_for_error($kwresults);
if (!defined($kwresults->{'occupation'}) || !scalar(@{ $kwresults->{'occupation'} })) {
  print "No relevant occupations were found.\n\n";
} else {
  print "Most relevant occupations for \"" . $kwquery . "\":\n";
  for my $occ (@{ $kwresults->{'occupation'} }) {
    print "  " . $occ->{'code'} . " - " . $occ->{'title'} . "\n";
  }
  print "\n";
}

exit;
1;
